apps/src/applab/reducers.js

```diff
@@ -0,0 +1,68 @@
+/** @file Redux reducer functions for App Lab.
+ *  @see http://redux.js.org/docs/basics/Reducers.html */
+// Strict linting: Absorb into global config when possible
+/* jshint
+ unused: true,
+ eqeqeq: true,
+ maxlen: 120
+ */
+'use strict';
+
+var ActionType = require('./actions').ActionType;
+var constants = require('./constants');
+var ApplabMode = constants.ApplabMode;
+
+var initialState = {
+  assetUrl: function () {},
+  isDesignModeHidden: undefined,
+  isEmbedView: undefined,
+  isReadOnlyWorkspace: undefined,
+  isShareView: undefined,
+  isViewDataButtonHidden: undefined,
+  mode: ApplabMode.CODE,
+  currentScreenId: null
+};
+
+function rootReducer(state, action) {
+  state = state || initialState;
+
+  switch (action.type) {
+    case ActionType.CHANGE_SCREEN:
+      if (state.currentScreenId === action.screenId) {
+        return state;
+      }
```

Comment:
I'm in progress on breaking this up, which may be overkill but i feel like modeling it now will demonstrate how to scale it later.